### DISCUSSION TITLE
How can I quickly find all TypeScript errors?

### DISCUSSION THREAD

**Discussion by justincy** [vercel/next.js#14234] [Help] [answered]
Next.js fails the build when it finds TypeScript errors. How can I find those errors without running `next build`? `next build` is slow and only reports one error at a time. VSCode only reports errors in files that are currently open.

**Comment by rafaelalmeidatk** [comment, marked as answer]
You can add a `type-check` command in the NPM scripts:

```
"scripts": {
  "type-check": "tsc",
  ...
},
```

This runs only the type-checking and will report all errors.

**Comment by jussihirvi** [comment]
Also you could open your project folder in VS Code. It will show TypeScript errors out of the box. Very practical. 

Using the tsc script probably requires that you install TypeScript globally, and then add the path to your PATH environment variable.

The script is actually not necessary, though it may make it easier to remember the procedure. You can also invoke tsc directly on the command line.
